public/src/overmind/actions.tsx

```diff
@@ -158,12 +174,18 @@ export const updateSubmission = async ({ state, effects }: Context, { owner, sub
         return
     }
 
-
     const clone = submission.clone()
     clone.status = status
     /* Update the submission status */
-    const result = await effects.grpcMan.updateSubmission(state.activeCourse, clone)
-    if (!success(result)) {
+    const response = await effects.api.client.updateSubmission({
+        courseID: state.activeCourse,
+        submissionID: submission.ID,
+        status: submission.status,
+        released: submission.released,
+        score: submission.score,
+    })
+
+    if (response.error) {
         /* If the update failed, revert the submission status */
```

Comment:
Is this accurate? There isn't any code that does the revert herein.

Also we should remove the blank line on 187.

Reply:
Removed. If the update fails, we do not update the state at all. The submission we've mutated on line 178 is a clone and can safely be ignored.